openedx/features/ucsd_features/validators.py

```diff
@@ -0,0 +1,125 @@
+
+
+class MasterPasswordValidator(object):
+    """
+    Validate all of the validators given in the validators list.
+
+    Parameters:
+        validators (list): the list of validators to validate along with their options.
+        min_validations (int): the num of minimum validation must pass for password.
+
+    Raises:
+        ValidationError if minimum number of validations is not passed.
+    """
+    def __init__(self, validators=[], min_validations=0):
+        self.check_validators_list(validators, min_validations)
+        self.validators_list = validators
+        self.validators = []
+        for validator in self.validators_list:
+            try:
+                validator_name = validator["NAME"]
+            except KeyError:
+                raise KeyError("Validator doesn't contains 'NAME'. Please enter valid Validator")
+            validator_options = validator.get("OPTIONS", {})
+            self.validators.append(import_string(validator_name)(**validator_options))
+        self.min_validations = min_validations
+
+    def validate(self, password, user=None):
+        errors, passed = [], 0
+        for v in self.validators:
+            try:
+                v.validate(password, user)
+            except ValidationError as e:
+                errors.append(e)
+            else:
+                passed += 1
+        if passed < self.min_validations:
+            errors.insert(0, ValidationError(" Fix at least {} of the following errors: ".format(
+                self.min_validations - passed)))
+            raise ValidationError(errors, code="too_few_validations",
+                                  params={"min_validations": self.min_validations})
+
+    def get_help_text(self):
+        text = ungettext(
+            "your password must contain at least %(min_validations)d of the following:",
+            "your password must contain at least %(min_validations)d of the followings:",
+            self.min_validations
+        ) % {"min_validations": self.min_validations}
+        for v in self.validators:
+            text += v.get_help_text()
+        return mark_safe(text)
+
+    def get_instruction_text(self):
+        text = ungettext(
+            "your password must contain at least %(min_validations)d of the following:",
+            "your password must contain at least %(min_validations)d of the followings:",
+            self.min_validations
+        ) % {"min_validations": self.min_validations}
+        for validator in self.validators:
+            if hasattr(validator, 'get_instruction_text'):
+                text += validator.get_instruction_text() + ","
+        return mark_safe(text[:-1])
+
+    @staticmethod
+    def check_validators_list(validators, min_validations):
+        if len(validators) >= min_validations >= 0:
+            return
+        elif len(validators) < min_validations:
+            raise Exception('Number of Validators in list is lower the the minimum number of required validations.')
```

Comment:
```suggestion
        if len(validators) < min_validations:
            raise Exception('Number of Validators in list is lower the the minimum number of required validations.')
```
From what I understand from the code, we only need a single if block.

Reply:
Done